persist/fs/commitlog/reader.go

```diff
@@ -193,6 +198,9 @@ func (r *reader) readLoop() {
 			if err == io.EOF {
 				eofFound = true
 			}
+			if err == nil {
+				data.IncRef()
```

Comment:
I would let the consumer IncRef, the Inc/Dec is mainly to signify ownership.  This method is passing ownership of `data` to someone else so it should not Inc and leave it `+1` as it's ownership has finished when it puts it onto a decoder chan.

Reply:
Ok cool that makes sense